dandi/upload.py

```diff
@@ -383,3 +398,67 @@ def check_replace_asset(
 
 def skip_file(msg: Any) -> Dict[str, str]:
     return {"status": "skipped", "message": str(msg)}
+
+
+def _bids_discover_and_validate(
+    dandiset_path: str,
+    paths: Optional[List[Union[str, Path]]] = None,
+    validation: Optional[str] = "require",
+) -> List[Path]:
+    """Temporary implementation for discovery and validation of BIDS datasets
+
+    References:
+    - unification of validation records: https://github.com/dandi/dandi-cli/issues/943
+    - validation "design doc": https://github.com/dandi/dandi-cli/pull/663
+    """
+    from .utils import find_files
+    from .validate import validate_bids
+
+    if paths:
+        bids_lookup_paths = set(paths)
+    else:
+        bids_lookup_paths = None
+    bids_descriptions = map(
+        Path, find_files(r"(^|[/\x5C])dataset_description\.json$", dandiset_path)
+    )
+    bids_datasets = [bd.parent for bd in bids_descriptions]
+    if bids_datasets:
+        lgr.debug(
+            "Detected %s under following paths: %s",
+            pluralize(len(bids_datasets), "BIDS dataset"),
+            ", ".join(str(i) for i in bids_datasets),
+        )
+
+    if validation != "skip":
+        if bids_lookup_paths:
+            bids_datasets_to_validate = list()
+            for p in bids_lookup_paths:
+                print(f"p:{p}")
```

Comment:
Are these `print`s just for develomental debugging?  If they're really needed, replace them with `lgr.debug()` calls with more informative messages.  Otherwise, get rid of them.

Reply:
there were these and some more traces of my debugging, purged them all and force-pushed.